lib/Config.js

```diff
@@ -506,6 +506,15 @@ function requestsConfigAssert(requestsConfig) {
             'config: invalid requests configuration. ' +
             'extractProtocolFromHeader must be set if viaProxy is ' +
             'set to true and must be a string');
+
+            // All headers in NodeJS are lowercase: to be exploitable
+            // we need to lowercase the value.
+            // eslint-disable-next-line no-param-reassign
+            requestsConfig.extractClientIPFromHeader =
+                requestsConfig.extractClientIPFromHeader.toLowerCase();
+            // eslint-disable-next-line no-param-reassign
+            requestsConfig.extractProtocolFromHeader =
+                requestsConfig.extractProtocolFromHeader.toLowerCase();
```

Comment:
They are not always present

```suggestion
            requestsConfig.extractClientIPFromHeader =
                requestsConfig.extractClientIPFromHeader?.toLowerCase();
            // eslint-disable-next-line no-param-reassign
            requestsConfig.extractProtocolFromHeader =
                requestsConfig.extractProtocolFromHeader?.toLowerCase();
```

Reply:
maybe this should be out of condition `viaProxy` as well